### DISCUSSION TITLE
Ability to create stubs without adding them to the autoloader

### DISCUSSION THREAD

**Discussion by mglaman** [rectorphp/rector#6623] [Q&A]
Is there a way to register stub files without adding them to the class mapping? The examples for Symfony, PHPUnit, etc add `stubs/*.php` and register them in the class map.

PHPStan does stubbing with `.stub` files and via configuration. Does Rector support this? Or is it only possible via autoloading?

We did a stub in `drupal-rector`:https://github.com/palantirnet/drupal-rector/blob/main/stubs/Drupal/Tests/BrowserTestBase.php. But this is registered before Drupal's PHPUnit bootstrap file, so it takes priority.

**Comment by TomasVotruba** [comment]
Hi, I think it should be possible via autoload section:
https://github.com/rectorphp/rector/blob/main/docs/static_reflection_and_autoload.md#static-reflection-and-autoload

These files are parsed to static reflection and should be available there.

If not, it still needs to be done probably.

**Reply by mglaman** [reply]
We need it to work without being autoloader before the actual Drupal source code. The reason the stub was added was for PHPUnit testing. 

I mistakenly thought it'd work like PHPStan which falls back on stub autoloading if the class is a miss.

I forgot to mention that, sorry. This stub is added purely for our PHPUnit tests.... Which means I should move the autoload declaration to autoload-dev 🤦‍♂️

Thanks. Thought through the solution while replying.

**Reply by TomasVotruba** [reply]
See the link and give it a try.
